jupyter_server/terminal/handlers.py

```diff
@@ -27,13 +27,26 @@ def get(self, *args, **kwargs):
     def on_message(self, message):
         super(TermSocket, self).on_message(message)
         self._update_activity()
+        self._set_state_busy()
 
     def write_message(self, message, binary=False):
         super(TermSocket, self).write_message(message, binary=binary)
         self._update_activity()
+        if message != '["stdout", "\\r\\n"]':
+            self._set_state_idle()
 
     def _update_activity(self):
         self.application.settings["terminal_last_activity"] = utcnow()
         # terminal may not be around on deletion/cull
         if self.term_name in self.terminal_manager.terminals:
             self.terminal_manager.terminals[self.term_name].last_activity = utcnow()
+
+    def _set_state_busy(self):
+        self.log.debug('set terminal execution_state as busy')
```

Comment:
Actually, we should only print this message if we aren't currently in busy state to avoid spurious messages.

Reply:
indeed, just print message on `set idle`